southbound/impl/src/main/java/org/opendaylight/ansible/southbound/AnsibleCommandServiceImpl.java

```diff
@@ -8,30 +8,150 @@
+import org.opendaylight.yang.gen.v1.urn.opendaylight.ansible.command.rev180821.commands.Command;
+import org.opendaylight.yang.gen.v1.urn.opendaylight.ansible.command.rev180821.commands.CommandBuilder;
+import org.opendaylight.yang.gen.v1.urn.opendaylight.ansible.command.rev180821.commands.CommandKey;
+import org.opendaylight.yangtools.yang.binding.InstanceIdentifier;
 import org.opendaylight.yangtools.yang.common.RpcResult;
 import org.opendaylight.yangtools.yang.common.RpcResultBuilder;
+import org.ops4j.pax.cdi.api.OsgiService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 @Singleton
 public class AnsibleCommandServiceImpl implements AnsibleCommandService {
     private static final Logger LOG = LoggerFactory.getLogger(AnsibleCommandServiceImpl.class);
+    public static final InstanceIdentifier<Command> ANSIBLE_COMMAND_PATH = InstanceIdentifier.create(Commands.class)
+            .child(Command.class);
+    private Map<Uuid, ManagedProcess> processMap = new HashMap<>();
+    private final DataBroker dataBroker;
 
     @Inject
-    public AnsibleCommandServiceImpl() {
-        LOG.info("constructor");
+    public AnsibleCommandServiceImpl(@OsgiService final DataBroker dataBroker) {
+        this.dataBroker = dataBroker;
     }
 
     @Override
     public ListenableFuture<RpcResult<RunAnsibleCommandOutput>> runAnsibleCommand(RunAnsibleCommandInput input) {
-        LOG.info("We made it!");
-        return RpcResultBuilder.success(new RunAnsibleCommandOutputBuilder().setStatus("OK").build()).buildFuture();
+        Uuid uuid;
+        String status;
+        try {
+            uuid = runAnsible(input.getHost(), input.getDirectory(), input.getFile());
+            status = "INPROGRESS";
+        } catch (ManagedProcessException e) {
+            status = "FAILED";
+            uuid = null;
+        }
+        return RpcResultBuilder.success(new RunAnsibleCommandOutputBuilder().setStatus(status).setUuid(uuid).build())
+                .buildFuture();
+    }
+
+    private Uuid runAnsible(String host, String dir, String file) throws ManagedProcessException {
+        Uuid uuid = new Uuid(UUID.randomUUID().toString());
+        LOG.info("Executing Ansible, new uuid for command is: " + uuid);
+        ManagedProcessBuilder ar = new ManagedProcessBuilder("ansible-runner").addArgument("-j").addArgument
+                ("--hosts").addArgument(host).addArgument("-p").addArgument(file);
+        ar.addArgument("run").addArgument(dir);
+        ar.setProcessListener(new AnsibleProcessListener(this, uuid));
+        ManagedProcess p = ar.build();
+        processMap.put(uuid, p);
+        LOG.info("Starting Ansible process");
+        try {
+            p.start();
+            LOG.info("Ansible Process is alive: {}", Boolean.toString(p.isAlive()));
+        } catch (ManagedProcessException e) {
+            LOG.warn("Process exited with error code: " + p.getProcLongName());
```

Comment:
do you want the stack trace here? Also use {} instead of +. Then just add a , e and the stack trace will be included

Reply:
will do